rewrite-gradle/build.gradle.kts

```diff
@@ -17,6 +17,8 @@ dependencies {
 
     implementation(project(":rewrite-properties"))
 
+    implementation("com.github.ben-manes.caffeine:caffeine:2.+")
```

Comment:
I don't think we want to add a dependency on caffeine here. I think it is ok just to cache the list of gradle distributions in memory indefinitely rather than relying on a TTL cache. We will pick up any new releases when the process is bounced.

Reply:
I don't want unexpected behavior when there are new gradle releases, regardless of how infrequent that might be. We do have a hard dependency on caffeine in `rewrite-maven`, so adding it to `rewrite-gradle` is not a big deal.

Reply:
wait, so you are saying a TTL cache is ok here?

Reply:
the suggested change was already a TTL cache with a TTL of forever ;)

Reply:
i also question why this is necessary at all. you would think this is one request per repo run.

Reply:
Fair enough